tests/stateful_test.py

```diff
@@ -1189,20 +1241,180 @@ def rule_rent_batch_721_1155(self, address):
             )
         ] = rentingd
 
+
+    def apply_stop_lend(self, renting: Renting):
+            lending_id = concat_lending_id(renting.nft_address, renting.token_id, renting.lending_id)
+            lending: Lending = self.lendings[lending_id]
+            if lending.will_auto_renew:
+                self.lendings[lending_id].available_amount += renting.rent_amount
+            else:
+                if lending.lend_amount == renting.rent_amount:
+                    del self.lendings[lending_id]
+                else:
+                    self.lendings[lending_id].lend_amount -= renting.rent_amount
+
+            del self.rentings[concat_renting_id(renting.nft_address, renting.token_id, renting.renting_id)]
```

Comment:
n  := lend amount
m := rented amount
l   := returned amount

let n >= m >= l

imagine that in fact m > l

in this case, we should not be deleting the renting, because the renter hasn't returned all the amounts yet, no?

Reply:
our contract does not support partial returns